flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -348,8 +348,13 @@ private void performAuthorization(AuthorizationServiceConfiguration serviceConfi
         }
 
         AuthorizationService authorizationService = allowInsecureConnections ? insecureAuthorizationService : defaultAuthorizationService;
-        Intent authIntent = authorizationService.getAuthorizationRequestIntent(authRequestBuilder.build());
-        mainActivity.startActivityForResult(authIntent, exchangeCode ? RC_AUTH_EXCHANGE_CODE : RC_AUTH);
+
+        try {
+            Intent authIntent = authorizationService.getAuthorizationRequestIntent(authRequestBuilder.build());
+            mainActivity.startActivityForResult(authIntent, exchangeCode ? RC_AUTH_EXCHANGE_CODE : RC_AUTH);
+        } catch (Exception ex) {
```

Comment:
Was able to reproduce this based on the steps you mentioned and did more research online where the API docs for AppAuth Android SDK confirms that it's thrown only when no suitable browser is found.  With that in mind, would you be able to make a change so that only the `ActivityNotFoundException` and introduce a new error code (e.g. `NO_BROWSER_AVAILABLE`)  and message (e.g. `No suitable browser is available`) that would that make it more easier to catch this specific issue on the Flutter/Dart side?

Reply:
That makes sense, changed that.